qtpy/tests/test_qtwidgets.py

```diff
@@ -130,3 +135,74 @@ def test_opengl_imports():
     QtPy makes QOpenGLWidget available in QtWidgets to maintain compatibility.
     """
     assert QtWidgets.QOpenGLWidget is not None
+
+@pytest.mark.skipif(sys.platform == 'darwin' and sys.version_info[:2] == (3, 7) and (PYQT5 or PYSIDE2), reason="test crashes on macOS CI with Python 3.7 with 'Illegal instruction: 4'")
+@pytest.mark.parametrize("keyword", ["dir", "directory"])
+@pytest.mark.parametrize("instance", [True, False])
+def test_qfiledialog_compat(tmp_path, qtbot, keyword, instance):
+    """
+    This function is testing if the decorators that renamed keyword are working.
+    It mas stop working if the Qt bindings do some overwriting of the methods in constructor. 
+    It should not happen, but we already meet that PySide team did similar things in the past 
+    (like overwriting enum module in PySide6==6.3.2).
+
+    keyword: str
+        The keyword that should be used in the function call.
+    instance: bool
+        If True, the function is called on the instance of the QFileDialog, otherwise on the class.
+    """
+    
+    class CloseThread(QtCore.QThread):
+        """
+        On some implementations the `getExistingDirectory` functions starts own 
+        event loop that will not trigger QTimer started before the call. Until the
+        dialog is closed the main event loop will be stopped.
+
+        Because of this it is required to use the thread to interact with the dialog.
```

Comment:
```suggestion
        On some implementations the `getExistingDirectory` functions starts own 
        event loop that will not trigger QTimer started before the call. Until
        the dialog is closed the main event loop will be stopped.

        Because of this it is required to use the thread to interact with the
        dialog.
```

80 columns.